### PR TITLE
feat: SLA report filter

### PR DESCRIPTION
**Pull request by iamsivin**
# Pull Request Template

## Description

This PR includes filtering SLA reports by agents, Inboxes, SLA policies, teams, and labels.

Fixes https://linear.app/chatwoot/issue/CW-3190/use-the-filter-components-in-the-reports

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

**Loom video**
https://www.loom.com/share/abb07e83bf7e4ac89abeb22802a0aff7?sid=22e7e36f-3a11-4582-b0be-3d22a63e0edf

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by scmmishra — CHANGES_REQUESTED** [review]
@iamsivin I moved the active filters as a separate component in `ActiveFilterChip.vue` tried to reduce the amount of template we have. See if you can do it for that for the main `Add filter` button too. Once done, it'll be easier to see how we can split the logic also

**Review by scmmishra — APPROVED** [review]

**Comment by scmmishra** [comment]
@iamsivin the changes look good for now. We can revaluate the filter components and see if the components can be simplified once we reuse it elsewhere

**Comment by muhsin-k** [comment]
@iamsivin Should we display the "Add Filter" button if all filters are selected? In my opinion, only the "Clear All" button is necessary in that case.
CleanShot 2024-04-12 at 08 38 47@2x

File: app/javascript/dashboard/routes/dashboard/settings/reports/components/SLA/SLAFilter.vue [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+        );
+        return {
+          id: item.id,
+          name: filterType === 'labels' ? item.title : item.name,
+          type: filterType,
+          options: buildFilterList(this[filterType], filterType),
+        };
+      });
+    },
+    hasActiveFilters() {
```
Comment (muhsin-k) [inline app/javascript/dashboard/routes/dashboard/settings/reports/components/SLA/SLAFilter.vue:156]: We can simplify it by directly checking if any of the `appliedFilters` has a value other than null. This approach eliminates the need to first compute `activeFilters` just to check if any filters are active.

```
hasActiveFilters() {
   return Object.values(this.appliedFilters).some(value => value !== null);
},
```

**Review by muhsin-k — APPROVED** [review]
LGTM

**Pull request merged**